File: src (cpp)/animal/reptilia/iguana.cpp
#include "iguana.h"

Iguana::Iguana(int _x, int _y, int _weight) : def_ID(10),
                                              def_ratio_meat(20),
                                              def_ratio_plant(40) {
  ID = def_ID;
  position.SetX(_x);
  position.SetY(_y);
  ratio_meat = def_ratio_meat;
  ratio_plant = def_ratio_plant;
  is_land_animal = true;
  is_water_animal = false;
  is_air_animal = false;
  weight = _weight;
}
void Iguana::Interact() {
  std::cout << "The iguana is calmly sleeping on a tree." << std::endl;
}

File: src/virtualzoo/animal/reptilia/Iguana.java
package virtualzoo.animal.reptilia;

import virtualzoo.animal.behavior.BehaviorTame;
import virtualzoo.animal.diet.Omnivora;

/**
 * Kelas Iguana mendefinisikan atribut untuk iguana.
 *
 * @author Rionaldi Chandraseta - 13515077
 * @version 3.0
 * @since 3.0
 */
public class Iguana extends Reptilia {

  /** <p>
   * Constructor
   *
   * Menciptakan iguana dengan posisi (locX,locY) dan berat animalWeight.
   * </p>
   * @param locX Nilai absis posisi.
   * @param locY Nilai ordinat posisi.
   * @param animalWeight Nilai berat iguana.
   */
  public Iguana(int locX, int locY, int animalWeight) {
    super();
    animalBehavior = new BehaviorTame();
    animalDiet = new Omnivora(animalWeight, 20, 40);
    id = 11;
    position.setLocation(locX, locY);
    isLandAnimal = true;
    isWaterAnimal = false;
    isAirAnimal = false;
  }

  /**
   * Mengatur interaksi dengan iguana.
   */
  public void interact() {
    System.out.println("The iguana is calmly sleeping on a tree.");
  }
}
